features/ui/src/main/java/org/opennms/alec/rest/SituationRestImpl.java

```diff
@@ -100,6 +100,26 @@ public SituationRestImpl(KeyValueStore<String> kvStore,
         client = new SituationClient(channel, new SituationToSituationProto(), canWeStore(kvStore));
     }
 
+    //Only for testing
+    protected SituationRestImpl(KeyValueStore<String> kvStore,
+                                SituationDatasource situationDatasource,
+                                AlarmDatasource alarmDatasource,
+                                RuntimeInfo runtimeInfo,
+                                GrpcConnectionConfig grpcConnectionConfig,
+                                SituationClient client) {
+        this.kvStore = Objects.requireNonNull(kvStore);
+        this.situationDatasource = Objects.requireNonNull(situationDatasource);
+        this.alarmDatasource = Objects.requireNonNull(alarmDatasource);
+        this.runtimeInfo = Objects.requireNonNull(runtimeInfo);
+        objectMapper = new ObjectMapper();
+
+        //channel to store situations
+        channel = ManagedChannelBuilder.forAddress(grpcConnectionConfig.getHost(), grpcConnectionConfig.getPort())
```

Comment:
I think  we should put all the gRPC and channel related logic inside of the `SituationClient` rather than have it bleed here into the REST implementation.